cdap-data-fabric/src/main/java/co/cask/cdap/data/view/ViewAdmin.java

```diff
@@ -43,9 +43,17 @@ public ViewAdmin(ViewStore store, ExploreFacade explore, ExploreTableNaming nami
 
   public boolean createOrUpdate(Id.Stream.View viewId, ViewSpecification spec) throws Exception {
     if (store.exists(viewId)) {
-      explore.disableExploreStream(viewId.getStream(), naming.getTableName(viewId));
+      ViewSpecification previousSpec = store.get(viewId);
+      if (previousSpec == null) {
```

Comment:
Do you still need to check for `exists` then? Isn't `null` previousSpec means it does not exist?